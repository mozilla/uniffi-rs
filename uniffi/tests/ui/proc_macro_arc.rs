use std::sync::Arc;

fn main() {}

pub struct Foo;

#[uniffi::export]
fn make_foo() -> Arc<Foo> {
    Arc::new(Foo)
}

mod child {
    use std::sync::Arc;

    enum Foo {}

    #[uniffi::export]
    fn take_foo(foo: Arc<Foo>) {
        match &*foo {}
    }
}

mod uniffi_types {
    pub use super::Foo;
}
