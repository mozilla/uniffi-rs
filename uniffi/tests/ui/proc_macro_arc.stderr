error[E0271]: type mismatch resolving `<Arc<child::Foo> as child::_::_::{closure#0}::TypeEq>::This == Arc<Foo>`
  --> tests/ui/proc_macro_arc.rs:18:22
   |
18 |     fn take_foo(foo: Arc<Foo>) {
   |                      ^^^^^^^^ type mismatch resolving `<Arc<child::Foo> as child::_::_::{closure#0}::TypeEq>::This == Arc<Foo>`
   |
note: expected this to be `Arc<Foo>`
  --> tests/ui/proc_macro_arc.rs:18:22
   |
18 |     fn take_foo(foo: Arc<Foo>) {
   |                      ^^^
   = note: enum `child::Foo` and struct `Foo` have similar names, but are actually distinct types
note: enum `child::Foo` is defined in module `crate::child` of the current crate
  --> tests/ui/proc_macro_arc.rs:15:5
   |
15 |     enum Foo {}
   |     ^^^^^^^^
note: struct `Foo` is defined in module `crate` of the current crate
  --> tests/ui/proc_macro_arc.rs:5:1
   |
5  | pub struct Foo;
   | ^^^^^^^^^^^^^^
note: required by a bound in `child::_::_::{closure#0}::assert_type_eq_all`
  --> tests/ui/proc_macro_arc.rs:18:22
   |
18 |     fn take_foo(foo: Arc<Foo>) {
   |                      ^^^
   |                      |
   |                      required by a bound in this
   |                      required by this bound in `child::_::_::{closure#0}::assert_type_eq_all`
   = note: this error originates in the macro `::uniffi::deps::static_assertions::assert_type_eq_all` (in Nightly builds, run with -Z macro-backtrace for more info)
